nix-libs/nodeLib/buildNodePackage.nix

```diff
@@ -22,6 +22,12 @@ let
                        attrValues concatStringsSep optionalString filter
                        optionalAttrs;
 
+  # Like `subtractLists` but tests each dependency element in the
+  # minuend list for membership in the subtrahend list using that
+  # dependency's `basicName`.
+  subtractDependencyLists = subtrahend:
+    filter (x: !(elem x.basicName subtrahend));
```

Comment:
Could using `basicName` run into trouble with namespaced packages?

Reply:
Good question. I need to check that.

Reply:
In fact, I don't like this very much, I think we should just use the full package and use `lib.subtractLists`...